### PR TITLE
Ran Provider (update) generator

### PR DESCRIPTION
**Pull request by juliancheal**
This keeps the provider inline with new updates for all providers.
Also removed rubocop_base.yml as no longer required.

### CODE REVIEW

**Comment by juliancheal** [comment]
@durandom @djberg96 @bronaghs I gone and broke it, any ideas why these changes have made the tests fail?

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -3,9 +3,10 @@
   SimpleCov.start
 end
 
-VCR.configure do |config|
-  config.ignore_hosts 'codeclimate.com' if ENV['CI']
-  config.cassette_library_dir = File.join(ManageIQ::Providers::Azure::Engine.root, 'spec/vcr_cassettes')
-end
```
Comment (durandom) [inline spec/spec_helper.rb]: We want to run vcr cassettes here.
Thats another thing that could be checked dynamically

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -3,9 +3,10 @@
...
+# Uncomment in case you use vcr cassettes
```
Comment (durandom) [inline spec/spec_helper.rb]: We want to run vcr cassettes here.
Thats another thing that could be checked dynamically

File: .travis.yml [hunk 3]
```diff
@@ -10,14 +10,5 @@ env:
...
-after_script: bundle exec codeclimate-test-reporter
-notifications:
-  webhooks:
-    urls:
-    - https://webhooks.gitter.im/e/73dcc0ceb8ba35a7a37c
-    on_success: change
-    on_failure: always
-    on_start: never
+install: bin/setup
+after_script: bundle exec codeclimate-test-reporter
```
Comment (durandom) [inline .travis.yml]: while you're at the generator you can add a newline here

File: .travis.yml [hunk 4]
```diff
@@ -10,14 +10,5 @@ env:
   - RUBY_GC_HEAP_GROWTH_FACTOR=1.25
 addons:
   postgresql: '9.4'
-before_install: bin/setup
-script:
-- bundle exec bin/rails app:test:providers:azure:setup app:test:providers:azure
-after_script: bundle exec codeclimate-test-reporter
-notifications:
```
Comment (durandom) [inline .travis.yml]: the hook shall stay

**Comment by durandom** [comment]
@juliancheal  why are new vcr cassettes part of this PR? 
I think besides the commented VCR path it was good to go.

**Comment by juliancheal** [comment]
It wasn't green and someone suggested re-running the cassettes.

**Comment by durandom** [comment]
@juliancheal ok. The problem now is, that you (or the generator) commented the VCR paths out and so CI does not use them. 

Please re-add the original cassettes and remove the vcr comments in spec/spec_helper.rb and it should be green

**Comment by juliancheal** [comment]
🍏

**Comment by juliancheal** [comment]
@durandom merge?

File: .travis.yml [hunk 5]
```diff
@@ -10,14 +10,5 @@ env:
   - RUBY_GC_HEAP_GROWTH_FACTOR=1.25
 addons:
   postgresql: '9.4'
-before_install: bin/setup
-script:
-- bundle exec bin/rails app:test:providers:azure:setup app:test:providers:azure
+install: bin/setup
 after_script: bundle exec codeclimate-test-reporter
-notifications:
```
Comment (durandom) [inline .travis.yml]: notifications should stay

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -3,9 +3,10 @@
   SimpleCov.start
 end
 
+# Uncomment in case you use vcr cassettes
 VCR.configure do |config|
-  config.ignore_hosts 'codeclimate.com' if ENV['CI']
-  config.cassette_library_dir = File.join(ManageIQ::Providers::Azure::Engine.root, 'spec/vcr_cassettes')
+ config.ignore_hosts 'codeclimate.com' if ENV['CI']
```
Comment (durandom) [inline spec/spec_helper.rb]: some extra space sneaked in, just remove this file from your commits

File: .travis.yml (hunk 5 above)
Comment (juliancheal) [inline .travis.yml]: Oh I thought I added it back in.

File: spec/spec_helper.rb (hunk 6 above)
Comment (juliancheal) [inline spec/spec_helper.rb]: I can't see the extra space?

**Comment by durandom** [comment]
@juliancheal looks good now, thanks for going through this till the end

**Pull request merged**